### PR TITLE
71 report populations per location

### PR DESCRIPTION
**Pull request by majabedi**
## Merge Request - GuideLine Checklist 

**Guideline** to check code before resolve WIP and approval, respectively.
As many checkboxes as possible should be ticked.

### Checks by code author:
* [ ] There is at least one issue associated with the pull request.
* [ ] The branch follows the naming conventions as defined in the [git workflow](git-workflow).
* [ ] New code adheres with the [coding guidelines](coding-guidelines)
* [ ] Tests for new functionality has been added
* [ ] A local test was succesful
* [ ] There is appropriate **documentation** of your work. (use doxygen style comments)
* [ ] If new third party software is used, did you pay attention to its license? Please remember to add it to the wiki after successful merging.
* [ ] If new mathematical methods or epidemiological terms are used, has the glossary been updated ? Did you provide further documentation ?
 is present or referenced. Please provide your references.
* [ ] The following questions are addressed in the documentation*:  Developers (what did you do?, how can it be maintained?), For users (how to use your work?), For admins (how to install and configure your work?)
* For documentation: Please write or update the Readme in the current working directory!

### Checks by code reviewer(s):
* [ ] Is the code clean of development artifacts e.g., unnecessary comments, prints, ...
* [ ] The ticket goals for each associated issue are reached or problems are clearly addressed (i.e., a new issue was introduced).
* [ ] There are appropriate **unit tests** and they pass.
* [ ] The git history is clean and linearized for the merge request.
* [ ] Coverage report for new code is acceptable.

### CODE REVIEW

**Comment by majabedi** [comment]
@dabele please try to review the code whenever you can.

File: cpp/models/abm/simulation.h [hunk 1]
```diff
@@ -58,13 +63,80 @@ class AbmSimulation
...
+        report_setting.report_statistics_per_location_type = true;
+    }
+
+    /**
+     * switch off the reporting for every single location.
+     */
+    void switch_off_reporting_for_every_location_type()
+    {
+        report_setting.report_statistics_per_location_type = false;
+    }
```
Comment (mknaranja) [inline cpp/models/abm/simulation.h:114]: I don't think we need switch on AND off. Use boolean input variable.

**Comment by xsaschako** [comment]
@mknaranja what's the status on this pull request? Should it be closed? Should it be a draft again and someone else needs to edit it?

**Comment by xsaschako** [comment]
Closed this in favor of a larger PR #425

**Pull request closed without merging**